src/decorators/string.ts

```diff
@@ -0,0 +1,180 @@
+    lowercase(): this;
+
+    /**
+     * Specifies the maximum length.
+     * @param length The maximum length.
+     */
+    max(length: number): this;
+
+    /**
+     * Specifies the minimum length.
+     * @param length The minimum length.
+     */
+    min(length: number): this;
+
+    /**
+     * Defines a pattern rule.
+     * @param pattern A regular expression object the string value must match against.
+     * @param name Optional name for patterns (useful with multiple patterns).
+     */
+    pattern(pattern: RegExp, name?: string): this;
+
+    /**
+     * Defines a pattern rule.
+     * @param pattern A regular expression object the string value must match against.
+     * @param name Optional name for patterns (useful with multiple patterns).
+     */
+    regex(pattern: RegExp, name?: string): this;
+
+    /**
+     * Replace characters matching the given pattern with the specified replacement string.
+     * @param pattern A regular expression object to match against, or a string of which
+     *   all occurrences will be replaced.
+     * @param replacement The string that will replace the pattern.
+     */
+    replace(pattern: RegExp, replacement: string): this;
+
+    /**
+     * Requires the string value to only contain a-z, A-Z, 0-9, and underscore _.
+     */
+    token(): this;
+
+    /**
+     * Requires the string value to contain no whitespace before or after.
+     * If the validation convert option is on (enabled by default), the string will be trimmed.
+     */
+    trim(): this;
+
+    /**
+     * Specifies that the string must be in uppercase.
+     */
+    uppercase(): this;
+
+    /**
+     * Requires the string value to be a valid RFC 3986 URI.
+     * @param options Optional settings:
+     *   scheme - Specifies one or more acceptable Schemes, should only include the scheme name.
+     *     Can be an Array or String (strings are automatically escaped for use in a Regular Expression).
+     *   allowRelative - Allow relative URIs. Defaults to false.
+     *   relativeOnly - Restrict only relative URIs. Defaults to false.
+     *   allowQuerySquareBrackets - Allows unencoded square brackets inside the query string.
+     *     This is NOT RFC 3986 compliant but query strings like abc[]=123&abc[]=456 are very
+     *     common these days. Defaults to false.
+     *   domain - Validate the domain component using the options specified in string.domain().
+     */
+    uri(options?: Joi.UriOptions): this;
+}
+
+export function getStringSchemaModifierProviders(getJoi: () => typeof Joi) {
+    const result: ModifierProviders<Joi.StringSchema, StringSchemaModifiers> = {
+        ...getAnySchemaModifierProviders(getJoi),
+        alphanum: () => ({ schema }) => schema.alphanum(),
+        creditCard: () => ({ schema }) => schema.creditCard(),
+        email: (options?: Joi.EmailOptions) => ({ schema }) => schema.email(options),
+        exactLength: (length: number) => ({ schema }) => schema.length(length),
```

Comment:
🙏 When we eventually update the README.md, let's remember to document any decorators that diverge from diverge from the Joi API. For example, we need to let consumers know to use `exactLength()` instead of `length`.